src/v2/stack_aggregator.py

```diff
@@ -120,6 +120,44 @@ def extract_user_stack_package_licenses(packages: NormalizedPackages):
     return get_license_service_request_payload(normalized_package_details)
 
 
+def _get_vulnerabilities(vulnerability_nodes):
+    """Get list of vulnerabilities associated with a package."""
+    public_vulns = []
+    private_vulns = []
+    for vuln in vulnerability_nodes:
+        if _is_private_vulnerability(vuln):
+            private_vulns.append(VulnerabilityFields(**_get_vulnerability_fields(vuln)))
+        else:
+            public_vulns.append(VulnerabilityFields(**_get_vulnerability_fields(vuln)))
+    return public_vulns, private_vulns
+
+
+def get_package_details(component: Dict) -> PackageDataWithVulnerabilities:
+    """Extract package details from given graph response."""
+    pkg_node = component.get("package", {})
+    version_node = component.get("version", {})
+    ecosystem, pkg = _get_pkg_from_graph_version_node(version_node)
+    github_details = get_github_details(pkg_node)
+    public_vulns, private_vulns = _get_vulnerabilities(component.get("vuln", {}))
+    recommended_latest_version = pkg_node.get("latest_non_cve_version", [""])[0]
+    licenses = version_node.get("declared_licenses", [])
+
+    latest_version = select_latest_version(
+        pkg.version,
+        pkg_node.get("libio_latest_version", [""])[0],
+        pkg_node.get("latest_version", [""])[0],
+        pkg.name
+    )
+    return pkg, PackageDataWithVulnerabilities(**pkg.dict(), ecosystem=ecosystem,
+                                               latest_version=latest_version,
+                                               github=github_details, licenses=licenses,
+                                               # (fixme) this is incorrect
```

Comment:
Can we fix `fixme` in new code?

Reply:
It is not possible as this value should be formulated during ingestion.